src/frontend/lib/utils.ts

```diff
@@ -35,18 +34,18 @@ export function promiseTimeout<T>(
   promise: Promise<T>,
   ms: number,
   msg?: string
-): Promise<T> {
-  let timeoutId: TimeoutID;
-  const timeout = new Promise((resolve, reject) => {
-    timeoutId = setTimeout(() => {
+) {
+  let timeoutId: number;
+  const timeout: Promise<void> = new Promise((_resolve, reject) => {
+    timeoutId = window.setTimeout(() => {
       reject(new Error(msg || "Timeout after " + ms + "ms"));
     }, ms);
   });
-  promise.finally(() => clearTimeout(timeoutId));
+  promise.finally(() => window.clearTimeout(timeoutId));
   return Promise.race([promise, timeout]);
 }
 
-export function parseVersionMajor(versionString?: string = "") {
+export function parseVersionMajor(versionString: string = "") {
```

Comment:
I dont know the context, why did this go from being optional to required?

Reply:
it's based on this TS error (i.e. the `?` isn't necessary):

![image](https://user-images.githubusercontent.com/18542095/147001834-666869dc-6289-493e-91b9-d28dd24f007a.png)